tests/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing.UnitTests/AnalyzerConfigFilesTests.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if !NETCOREAPP1_1 && !NET46
+
+using System;
+using System.Collections.Immutable;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Text;
+using Xunit;
+using CSharpTest = Microsoft.CodeAnalysis.Testing.TestAnalyzers.CSharpAnalyzerTest<
+    Microsoft.CodeAnalysis.Testing.AnalyzerConfigFilesTests.HighlightBracesIfAnalyzerConfigMissingAnalyzer>;
+
+namespace Microsoft.CodeAnalysis.Testing
+{
+    public class AnalyzerConfigFilesTests
+    {
+        private const string RootEditorConfig = @"
+root = true
+
+[*]
+key = value
+";
+
+        [Fact]
+        public async Task TestDiagnosticInNormalFile()
+        {
+            await new CSharpTest
+            {
+                TestState =
+                {
+                    Sources = { "namespace MyNamespace { }" },
+                    ExpectedDiagnostics = { new DiagnosticResult(HighlightBracesIfAnalyzerConfigMissingAnalyzer.Descriptor).WithLocation(1, 23) },
+                    AnalyzerConfigFiles =
+                    {
+                        ("/.editorconfig", SourceText.From(RootEditorConfig, Encoding.UTF8)),
+                    },
```

Comment:
Wondering when this can be different than using AdditionalFiles like https://github.com/dotnet/roslyn-analyzers/blob/1ead3c96987ec6200209ab602529b9bab573833d/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/DoNotPrefixEnumValuesWithTypeNameTests.cs#L172 for example?

Reply:
> Wondering when this can be different than using AdditionalFiles

An **.editorconfig** which is added only as an additional file is meaningless on its own. The only reason that analyzer works is it [manually parses the file](https://github.com/dotnet/roslyn-analyzers/blob/3fb878177fb161344a57784957485450544e0f84/src/Utilities/Compiler/Options/AnalyzerOptionsExtensions.cs#L662-L678).

Reply:
@shawell Ahhhh, that's why it works in roslyn-analyzers tests without a `[*.cs]` section while the compiler requires it.